File: 哈希表/36. 有效的数独/有效的数独.go
func isValidSudoku(board [][]byte) bool {
	col, box, row := make([][10]int, 9), make([][10]int, 9), make([][10]int, 9)
	for i := 0; i < 9; i++ {
		for j := 0; j < 9; j++ {
			if board[i][j] == '.' {
				continue
			}
			number := int(board[i][j] - '0')
			if row[i][number] == 1 || col[j][number] == 1 || box[i/3*3+j/3][number] == 1 {
				return false
			}
			row[i][number] = 1
			col[j][number] = 1
			box[i/3*3+j/3][number] = 1
		}
	}
	return true
}

File: 哈希表/36. 有效的数独/有效的数独.cpp
#include<iostream>
#include<vector>
using namespace std;
// 上面的头文件与题解无光,只是为了在本地编译器里不报错

class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        vector<vector<bool>> row(9, vector<bool>(10, false));
        vector<vector<bool>> col(9, vector<bool>(10, false));
        vector<vector<bool>> box(9, vector<bool>(10, false));
        for (int i = 0; i < 9; ++i) {
            for (int j = 0; j < 9; ++j) {
                if (board[i][j] != '.') {
                    int num = board[i][j] - '0';
                    if (row[i][num] || col[j][num] || box[i / 3 * 3 + j / 3][num])return false;
                    row[i][num] = true;
                    col[j][num] = true;
                    box[i / 3 * 3 + j / 3][num] = true;
                }
            }
        }
        return true;
    }
};